cpp/benchmarks/io/cuio_generate_benchmark_input.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/table/table.hpp>
+
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+#include <memory>
+#include <random>
+#include <vector>
+
+template <typename T>
+constexpr auto stddev()
+{
+  // Columns of wider types -> larger range of values
+  return 1l << (sizeof(T) * 4);
+}
+
+// Type trait for  cudf timestamp types
+template <typename>
+struct is_timestamp {
+  constexpr static bool value = false;
+};
+
+template <typename T>
+struct is_timestamp<cudf::detail::timestamp<T>> {
+  constexpr static bool value = true;
+};
+
+// nanoseconds in the type's unit
+template <typename Ts>
+constexpr int64_t nanoseconds();
+
+template <>
+constexpr int64_t nanoseconds<cudf::timestamp_ns>()
+{
+  return 1;
+}
+
+template <>
+constexpr int64_t nanoseconds<cudf::timestamp_us>()
+{
+  return 1000l;
+}
+
+template <>
+constexpr int64_t nanoseconds<cudf::timestamp_ms>()
+{
+  return 1000l * 1000;
+}
+
+template <>
+constexpr int64_t nanoseconds<cudf::timestamp_s>()
+{
+  return 1000l * 1000 * 1000;
+}
+
+template <>
+constexpr int64_t nanoseconds<cudf::timestamp_D>()
+{
+  return 24l * 60 * 60 * nanoseconds<cudf::timestamp_s>();
```

Comment:
Would it be possible to use the ratio programatically. Then all logic can live in a single template rather than specializations.

Something like 
```c++
using ratio = std::ratio_divide<T::period, timestamp_ns::period>;
return ratio::num/ratio::den;
```
(I'm not sure if this is the right syntax, but you get the idea)

Reply:
I had no idea this existed! It's so much better now, thanks for the suggestion :)